pkg/image/registry/imagestreammapping/rest.go

```diff
@@ -90,17 +99,38 @@ func (s *REST) Create(ctx kapi.Context, obj runtime.Object) (runtime.Object, err
 		Image:                image.Name,
 	}
 
-	if !api.AddTagEventToImageStream(stream, tag, next) {
-		// nothing actually changed
-		return &unversioned.Status{Status: unversioned.StatusSuccess}, nil
-	}
-
-	api.UpdateTrackingTags(stream, tag, next)
-
-	if _, err := s.imageStreamRegistry.UpdateImageStreamStatus(ctx, stream); err != nil {
+	err = wait.ExponentialBackoff(wait.Backoff{Steps: maxRetriesOnConflict}, func() (bool, error) {
+		lastEvent := api.LatestTaggedImage(stream, tag)
+		if !api.AddTagEventToImageStream(stream, tag, next) {
+			// nothing actually changed
+			return true, nil
+		}
+		api.UpdateTrackingTags(stream, tag, next)
+		_, err := s.imageStreamRegistry.UpdateImageStreamStatus(ctx, stream)
+		if err == nil {
+			return true, nil
+		}
+		if !errors.IsConflict(err) {
+			return true, err
```

Comment:
Not sure that it matters- ExponentialBackoff will return due to err != nil. Technically we're also "done" at this point...